storage/core/src/types.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2022-2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Database backend support building blocks
+
+use core::ops::Range;
+
+/// Used to identify particular key-value map in the database
+#[derive(Eq, PartialEq, PartialOrd, Ord, Clone, Copy, Debug)]
+pub struct DbMapId(usize);
+
+impl DbMapId {
+    /// New index
+    pub const fn new(id: usize) -> Self {
+        DbMapId(id)
```

Comment:
Super tiny nitpick: Self also can be used here. 🙃 
```suggestion
        Self(id)
```